struqture/src/fermions/fermionic_indices.rs

```diff
@@ -727,10 +727,26 @@ impl ModeIndex for HermitianFermionProduct {
             false => return Err(StruqtureError::IncorrectlyOrderedIndices),
         }
 
-        if creators.iter().next() > annihilators.iter().next() {
+        let mut number_equal_indices = 0;
+
+        for (creator, annihilator) in creators.iter().zip(annihilators.iter()) {
+            match annihilator.cmp(creator) {
+                std::cmp::Ordering::Less => {
+                    return Err(StruqtureError::CreatorsAnnihilatorsMinimumIndex {
+                        creators_min: Some(*creator),
+                        annihilators_min: Some(*annihilator),
+                    });
+                }
+                std::cmp::Ordering::Greater => break,
+                _ => {
+                    number_equal_indices += 1;
+                }
+            }
+        }
+        if creators.len() > number_equal_indices && annihilators.len() == number_equal_indices {
             return Err(StruqtureError::CreatorsAnnihilatorsMinimumIndex {
-                creators_min: creators.iter().min().copied(),
-                annihilators_min: annihilators.iter().min().copied(),
+                creators_min: None,
```

Comment:
Also here seems they should be swapped. Also, I think it should be `nth(number_equal_indices)` because `nth(N)` already returns the Nth + 1 element.

Reply:
Fixed